src/store/token-dashboard/sagas.ts

```diff
@@ -335,28 +657,55 @@ function* connectWallet() {
 function* removeWallet(action: ConfirmRemoveWalletAction) {
   try {
     const removeWallet = action.payload.wallet
+    const removeChain = action.payload.chain
     const accountUserId: Nullable<ID> = yield select(getUserId)
     const userMetadata: ReturnType<typeof getAccountUser> = yield select(
       getAccountUser
     )
     const updatedMetadata = newUserMetadata({ ...userMetadata })
 
-    const currentAssociatedWallets: Record<string, any> = yield call(
-      AudiusBackend.fetchUserAssociatedWallets,
-      updatedMetadata
-    )
-    if (
-      currentAssociatedWallets &&
-      !(removeWallet in currentAssociatedWallets)
-    ) {
-      // The wallet already exists in the assocaited wallets set
-      yield put(updateWalletError({ errorMessage: 'Unable to remove wallet' }))
-      return
-    }
+    if (removeChain === 'eth') {
+      const currentAssociatedWallets: Record<string, any> = yield call(
+        AudiusBackend.fetchUserAssociatedEthWallets,
+        updatedMetadata
+      )
+      if (
+        currentAssociatedWallets &&
+        !(removeWallet in currentAssociatedWallets)
+      ) {
+        // The wallet already exists in the assocaited wallets set
```

Comment:
associated

Reply:
also, wouldn't `!(removeWallet in currentAssociatedWallets)` mean it's _not_ in the associated wallets set? Should this comment say it _doesn't_ exist in the associated wallet set?